packages/gatsby/src/redux/actions/public.js

```diff
@@ -35,6 +35,16 @@ const findChildrenRecursively = (children = []) => {
   return children
 }
 
+const getCommonDir = (path1, path2) => {
+  var iChar
+  for (iChar = 0; iChar < path1.length; iChar += 1) {
+    if (path1[iChar] !== path2[iChar]) {
+      return path1.substring(0, iChar)
+    }
+  }
+  return path1
+}
```

Comment:
Comparing paths character by character is potentially flaky:
 - there might be weirdness on Windows if one path contains slashes and other path contains backslashes (tho to be fair, we shouldn't hit this code path on Windows ever (`/` being inaccessible).
 - There might be casing difference that would cause characters to differ, but still should be considered same path.

I'll try to work it out, so we can merge this today